From 82b7ee2c64f8a9eb815d8c0f82c392c2565c1bf8 Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Fri, 18 Oct 2019 15:08:18 -0400
Subject: [PATCH 10/11] Fixes for native bridge wrapper

- Be more specific about houdini files to delete (avoid deleting
  bundled system/etc/houdini*.sfs)
- Use PRODUCT_PROPERTY_OVERRIDES for ro.dalvik.vm.native.bridge to
  correctly override value from build/make/target/product/runtime_libart.mk
- Use losetup/mount to mount the houdini sfs image, toybox's mount does
  not seem to be able to mount -o loop... properly
---
 nativebridge/Android.mk              | 2 +-
 nativebridge/bin/enable_nativebridge | 3 ++-
 nativebridge/nativebridge.mk         | 2 +-
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/nativebridge/Android.mk b/nativebridge/Android.mk
index 378ce69..0e4922d 100644
--- a/nativebridge/Android.mk
+++ b/nativebridge/Android.mk
@@ -20,7 +20,7 @@ LOCAL_SHARED_LIBRARIES := libcutils libdl liblog
 LOCAL_C_INCLUDES := system/core/libnativebridge/include
 LOCAL_MULTILIB := both
 LOCAL_POST_INSTALL_CMD := $(hide) \
-    rm -rf $(TARGET_OUT)/*/{arm*,*houdini*} {$(TARGET_OUT),$(PRODUCT_OUT)}/vendor/{*/arm*,*/*houdini*}; \
+    rm -rf $(TARGET_OUT)/{bin,lib}/{arm*,*houdini*} {$(TARGET_OUT),$(PRODUCT_OUT)}/vendor/{bin,lib}/{arm*,*houdini*}; \
     mkdir -p $(TARGET_OUT)/{lib/arm,$(if $(filter true,$(TARGET_IS_64_BIT)),lib64/arm64)}; \
     touch $(TARGET_OUT)/lib/libhoudini.so $(if $(filter true,$(TARGET_IS_64_BIT)),$(TARGET_OUT)/lib64/libhoudini.so)
 
diff --git a/nativebridge/bin/enable_nativebridge b/nativebridge/bin/enable_nativebridge
index afe2115..ddc5486 100644
--- a/nativebridge/bin/enable_nativebridge
+++ b/nativebridge/bin/enable_nativebridge
@@ -24,7 +24,8 @@ fi
 if [ -s /system/lib$1/libhoudini.so ]; then
 	log -pi -thoudini "found /system/lib$1/libhoudini.so"
 elif [ -e /system/etc/houdini$v.sfs ]; then
-	mount /system/etc/houdini$v.sfs $dest_dir
+	lodevice=`losetup -r -f -s /system/etc/houdini$v.sfs`
+	[ -n "$lodevice" ] && mount $lodevice $dest_dir
 else
 	if mountpoint -q $dest_dir; then
 		kill -9 `fuser -m $dest_dir`
diff --git a/nativebridge/nativebridge.mk b/nativebridge/nativebridge.mk
index 6f0a236..1fa831e 100644
--- a/nativebridge/nativebridge.mk
+++ b/nativebridge/nativebridge.mk
@@ -32,7 +32,7 @@ PRODUCT_PROPERTY_OVERRIDES += \
 endif
 
 ifneq ($(HOUDINI_PREINSTALL),intel)
-PRODUCT_DEFAULT_PROPERTY_OVERRIDES := ro.dalvik.vm.native.bridge=libnb.so
+PRODUCT_PROPERTY_OVERRIDES := ro.dalvik.vm.native.bridge=libnb.so
 
 PRODUCT_PACKAGES := libnb
 endif
-- 
2.21.0

